src/modules/rewards/services/arb-rebate-service.ts

```diff
@@ -36,12 +41,110 @@ export class ArbRebateService {
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
     private marketPriceService: MarketPriceService,
     private ethProvidersService: EthProvidersService,
     private appConfig: AppConfig,
     private dataSource: DataSource,
   ) {}
 
+  public async getEarnedRewards(userAddress: string) {
+    userAddress = assertValidAddress(userAddress);
+
+    const data = await this.cacheManager.get(getArbEarnedRewardsCacheKey(userAddress));
+    if (data) return data;
+
+    const baseQuery = this.buildBaseQuery(this.arbRewardRepository.createQueryBuilder("r"), userAddress);
+    const { arbRewards } = await baseQuery
+      .select("SUM(CAST(r.amount as DECIMAL))", "arbRewards")
+      .getRawOne<{ arbRewards: string }>();
+
+    if (this.appConfig.values.app.cacheDuration.rebatesData) {
+      await this.cacheManager.set(
+        getArbEarnedRewardsCacheKey(userAddress),
+        arbRewards,
+        this.appConfig.values.app.cacheDuration.rebatesData,
+      );
+    }
+
+    return arbRewards;
+  }
+
+  public async getArbRebatesSummary(userAddress: string) {
+    userAddress = assertValidAddress(userAddress);
+
+    let data = await this.cacheManager.get(getArbRebatesSummaryCacheKey(userAddress));
+    if (data) return data;
+
+    const baseQuery = this.buildBaseQuery(this.arbRewardRepository.createQueryBuilder("r"), userAddress);
+    baseQuery.andWhere("r.isClaimed = :isClaimed", { isClaimed: false });
+    const [{ depositsCount }, { unclaimedRewards }, { volumeUsd }] = await Promise.all([
+      baseQuery.select("COUNT(DISTINCT r.depositPrimaryKey)", "depositsCount").getRawOne<{
+        depositsCount: string;
+      }>(),
+      baseQuery.select("SUM(CAST(r.amount as DECIMAL))", "unclaimedRewards").getRawOne<{
+        unclaimedRewards: number;
+      }>(),
+      baseQuery
+        .leftJoinAndSelect("r.deposit", "d")
+        .leftJoinAndSelect("d.token", "t")
+        .leftJoinAndSelect("d.price", "p")
+        .select(`COALESCE(SUM(d.amount / power(10, t.decimals) * p.usd), 0)`, "volumeUsd")
+        .getRawOne<{
+          volumeUsd: number;
+        }>(),
+    ]);
+
+    data = {
+      depositsCount: parseInt(depositsCount),
+      unclaimedRewards,
+      volumeUsd,
+      claimableRewards: "0",
+    };
+
+    if (this.appConfig.values.app.cacheDuration.rebatesData) {
+      await this.cacheManager.set(
+        getArbRebatesSummaryCacheKey(userAddress),
+        data,
+        this.appConfig.values.app.cacheDuration.rebatesData,
+      );
+    }
+
+    return data;
+  }
+
+  public async getArbRebateRewards(query: GetRewardsQuery) {
+    const limit = parseInt(query.limit ?? "10");
+    const offset = parseInt(query.offset ?? "0");
+    const userAddress = assertValidAddress(query.userAddress);
+
+    const baseQuery = this.buildBaseQuery(this.arbRewardRepository.createQueryBuilder("r"), userAddress);
+
+    const rewardsQuery = baseQuery
+      .leftJoinAndSelect("r.rewardToken", "rewardToken")
```

Comment:
Is this left join needed?

Reply:
No! Good catch, I've removed it from this file and also from op-rebate-service